cmd/prometheus/main.go

```diff
@@ -668,6 +668,10 @@ func main() {
 	}
 	if cfgFile.StorageConfig.TSDBConfig != nil {
 		cfg.tsdb.OutOfOrderTimeWindow = cfgFile.StorageConfig.TSDBConfig.OutOfOrderTimeWindow
+		if cfgFile.StorageConfig.TSDBConfig.Retention != nil {
+			cfg.tsdb.RetentionDuration = cfgFile.StorageConfig.TSDBConfig.Retention.Time
+			cfg.tsdb.MaxBytes = cfgFile.StorageConfig.TSDBConfig.Retention.Size
```

Comment:
It's possible that one is set via CLI and one via config file. So we should check both for >0 before overriding.